test/compare_results.py

```diff
@@ -0,0 +1,97 @@
+#!/usr/bin/env python3
+
+import sys
+import argparse
+from pytest import approx
+
+QUANTUM_ENERGY_IDX=4
+
+def find_start_end_block(lines, blockname):
+    """Identifies start and end of blocks"""
+    for i, line in enumerate(lines):
+        if "Start " + blockname in line:
+            start_idx = i
+        elif "End " + blockname in line:
+            end_idx = i
+    return start_idx, end_idx
+
+def extract_quantum_block(lines):
+    """Extracts only the quantum block from a chk file"""
+    idxs = find_start_end_block(lines, "Quantum")
+    temp = lines[idxs[0]+4:idxs[-1]]
+    return [line.split() for line in temp]
+
+def extract_quantum_energies(block):
+    return [float(line[QUANTUM_ENERGY_IDX]) for line in block]
+
+def read_chk_file(fname):
+    with open(fname, 'r') as fp:
+        lines = fp.readlines()
+        if lines:
+            lines  = filter( lambda x : x != '\n', lines ) # remove empty lines
+            lines  = filter( lambda x : '<-' not in x, lines ) # remove commented lines
+    return list(lines)
+
+def read_energy_column(fname, column_no):
+    lines = read_chk_file(fname)
+    lines = lines[:-1] # remove last line
+    return [float(line.split()[column_no]) for line in lines]
```

Comment:
Very minor suggestion (unless you want to explicitly have the removal of the last line as a step)

```suggestion
    return [float(line.split()[column_no]) for line in lines[:-1]]
```

Reply:
Yes, that's explicit. I've improved that (hilariously useless) comment though...